From 324e5090aaac13a2896a2e22a287583ad8f00969 Mon Sep 17 00:00:00 2001
From: Kazuhito Hagio <k-hagio-ab@nec.com>
Date: Wed, 23 Dec 2020 16:38:41 +0900
Subject: [PATCH 04/13] extensions/eppic.mk: move ping check to recipe script

Without this patch, in an environment where ping to github.com does
not work, "make clean" at the top-level crash directory always takes
about 10 seconds unnecessarily.

  $ time make clean
  ...
  real    0m10.398s

Signed-off-by: Kazuhito Hagio <k-hagio-ab@nec.com>
Signed-off-by: Lianbo Jiang <lijiang@redhat.com>
---
 extensions/eppic.mk | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/extensions/eppic.mk b/extensions/eppic.mk
index c79170a596b7..bda69da6706f 100644
--- a/extensions/eppic.mk
+++ b/extensions/eppic.mk
@@ -24,7 +24,6 @@ ifeq ($(TARGET), X86)
 endif
 
 APPFILE=eppic/applications/crash/eppic.c
-GITHUB := $(shell ping -c 1 github.com | grep "1 received")
 GIT := $(shell which git 2> /dev/null)
 
 all:
@@ -38,7 +37,7 @@ all:
              if [ -n "$(EPPIC_GIT_URL)" ]; then \
                git clone "$(EPPIC_GIT_URL)" eppic; \
              else \
-	          if [ -n "$(GITHUB)" ] ; then \
+	          if ping -c 1 -W 5 github.com >/dev/null ; then \
 		    git clone https://github.com/lucchouina/eppic.git eppic; \
 	          fi; \
              fi; \
-- 
2.17.1

